crates/net/network/src/transactions/fetcher.rs

```diff
@@ -898,12 +898,24 @@ impl Hash for TxSizeMetadata {
     }
 }
 
+impl TxSizeMetadata {
+    fn get_size(&self) -> Option<usize> {
+        if self.tx_encoded_len == 0 {
+            None
+        } else {
+            Some(self.tx_encoded_len)
+        }
+    }
+}
+
+
 /// Metadata of a transaction hash that is yet to be fetched.
 #[derive(Debug, Constructor)]
 pub(super) struct TxFetchMetadata {
     /// The number of times a request attempt has been made for the hash.
     retries: u8,
     /// Peers that have announced the hash, but to which a request attempt has not yet been made.
+
     fallback_peers: LruCache<PeerId>,
```

Comment:
some recent investigation has lead us to remove this dependency from the codebase...instead please use `LruMap<PeerId, ()>`

Reply:
Gotcha, will do.

Reply:
another pr is addressing this https://github.com/paradigmxyz/reth/pull/7312 by changing `LruCache` internally, so it's ok to reuse it here